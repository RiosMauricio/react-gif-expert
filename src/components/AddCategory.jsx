import { useState } from "react"

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('')

    const onSubmit = (e) => {
        e.preventDefault();
        const cleanValue = inputValue.trim(); 

        if (cleanValue.length < 1) return 

        // setCategories((categories) => [...categories, inputValue]);
        onNewCategory(cleanValue);
        setInputValue('');
    }

    return (
        <>
            <form onSubmit={onSubmit}>
                <input
                    type="text"
                    placeholder="Buscar GIFs"
                    value={inputValue}
                    onChange={(e) => setInputValue(e.target.value)}
                />
            </form>
        </>
    )
}
