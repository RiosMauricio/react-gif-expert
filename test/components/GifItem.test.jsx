import { render, screen } from "@testing-library/react"
import { GifItem } from '../../src/components/GifItem'

describe('Pruebas en GifItem', () => {
    const title = 'Messi';
    const url = 'https://youtube.com/';

    test('debe hacer match con snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el url y alt indicado', () => {
        render(<GifItem title={title} url={url} />);
        // expect(screen.getByRole('img').src).toBe(url);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
})