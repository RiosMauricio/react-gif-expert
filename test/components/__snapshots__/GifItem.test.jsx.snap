// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem debe hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Messi"
      src="https://youtube.com/"
    />
    <p>
      Messi
    </p>
  </div>
</div>
`;
